### DISCUSSION TITLE
Merge BoundTo with Bound annotation

### DISCUSSION THREAD

**Discussion by badoualy** [sczerwinski/android-hilt#131] [Ideas]
Deprecate BoundTo (and delete it at some point) and merge it into `Bound`
supertype can be nullable, and if null will behave as `Bound` currently would.

Maybe I'm missing a reason why you made two distinct annotations

**Comment by sczerwinski** [comment]
Actually, when making `supertype` nullable, I'm getting an error:
>  An annotation parameter cannot be nullable

I could theoretically make the default supertype `Any::class` or `Nothing::class`, and treat it as `null`, but this is not ideal:
1. It would no longer be possible to actually bind something to `Any` or `Nothing`.
2. It wouldn't look great semantically.

This topic was also discussed here: https://github.com/sczerwinski/android-hilt/issues/8#issuecomment-751363270

The other possible workaround would be:
```kotlin
val supertype: Array<KClass<*>> = []
```
but then, multiple supertypes could be defined, which I'm not sure makes sense.

**But I'm still open for suggestions.**
